### ISSUE TITLE
Protocol incorrectly matches when positional or keyword parameter matches *args

### ISSUE THREAD

**Issue by danielknell** [python/mypy#21014]
**Bug Report**

When a protocol defines a function with a keyword or positional parameter `(a: int) -> None`, a concrete function where said parameter is missing, but matches the `*args` parameter `(*args: int) -> None` will be seen as implementing the protocol.

I would expect `(a: int) -> None` to also need matching `**kwargs` such as `(*args: int, **kwargs: int) -> None` and for it to only match `*args` alone when specified as a positional only argument (e.g. `(a: int, /) -> None`)

Otherwise it cannot be called safely, `pyright` and `ty` both successfully raise errors for this case.

**To Reproduce**

https://mypy-play.net/?mypy=master&python=3.14&flags=strict&gist=a5f86e32e70cbc84c45a53db5426f78f

**Expected Behavior**

```
main.py:13: error: Argument 1 to "f" has incompatible type "C"; expected "P"  [arg-type]
```

because calling the method as `foo(a=1)` would fail.

**Actual Behavior**

mypy:
```
Success: no issues found in 1 source file
```

pyright:
```
Argument of type "C" cannot be assigned to parameter "o" of type "P" in function "f"
  "C" is incompatible with protocol "P"
    "foo" is an incompatible type
      Type "(*args: int) -> None" is not assignable to type "(a: int) -> None"
        Missing keyword parameter "a"  (reportArgumentType)
```

ty:
```
Argument to function `f` is incorrect: Expected `P`, found `C` (invalid-argument-type) [Ln 17, Col 3]
```

python:
```
Traceback (most recent call last):
  File "xxx.py", line 17, in <module>
    f(C())
    ~^^^^^
  File "xxx.py", line 14, in f
    o.foo(a=1)
    ~~~~~^^^^^
TypeError: C.foo() got an unexpected keyword argument 'a'
```

**Your Environment**

- Mypy version used: 1.19.1 and master
- Mypy command-line flags: `--strict`
- Mypy configuration options from `mypy.ini` (and other config files):
- Python version used: `1.14`

**Comment by ilevkivskyi** [comment]
I am quite sure this is a duplicate, but can't find original after a quick search. This is a very old intentional exception that we should get rid of at some point, at least with `--strict`. This is not specific to protocols, nor to star arguments, mypy currently allows something like this:
```python
class B:
    def foo(self, x: int) -> None: ...
class C(B):
    def foo(self, arg: int) -> None: ...
```
even though it is not safe. This is because of tons of code written in days predating mypy.

**Comment by hauntsaninja** [comment]
For LSP, I had opened https://github.com/python/mypy/pull/18355. Based on primer it will need to be off by default.

The code paths for LSP and this are different though, so that PR won't fix this.

**Comment by ilevkivskyi** [comment]
Yeah. Btw for protocols it is also just removing `ignore_pos_arg_names=True` from `is_protocol_implementation()`. Btw I _guess_ fallout may be smaller for protocols.
